core-backend/wasmtime/src/funcs.rs

```diff
@@ -829,12 +829,9 @@ where
 
     pub fn forbidden(store: &mut Store<StoreData<E>>) -> Func {
         let func = move |mut caller: Caller<'_, StoreData<E>>| -> Result<(), Trap> {
-            caller.data_mut().termination_reason = Some(TerminationReason::Trap {
-                explanation: Some(TrapExplanation::Other(
-                    "Unable to call a forbidden function".into(),
-                )),
-                description: None,
-            });
+            caller.data_mut().termination_reason = Some(TerminationReason::Trap(
+                TrapExplanation::Other("Unable to call a forbidden function".into()),
+            ));
```

Comment:
Probably better to provide `TerminationReason::Trap(TrapExplanation::ForbiddenFunction)`?


By the way, for `TrapExplanation` `Cow` seems good, to don't allocate until we need to do something with the inner str